src/modules/keyboard_layout.rs

```diff
@@ -0,0 +1,130 @@
+use crate::{config::KeyboardLayoutModule, style::HeaderButtonStyle};
+use hyprland::{
+    ctl::switch_xkb_layout::SwitchXKBLayoutCmdTypes, event_listener::AsyncEventListener,
+    shared::HyprData,
+};
+use iced::{
+    subscription::channel,
+    theme::Button,
+    widget::{button, text},
+    Element, Subscription,
+};
+use log::{debug, error};
+use std::{
+    any::TypeId,
+    sync::{Arc, RwLock},
+};
+
+fn get_layout() -> Layout {
+    let active = hyprland::data::Devices::get()
+        .ok()
+        .and_then(|devices| {
+            devices
+                .keyboards
+                .iter()
+                .find(|k| k.main)
+                .map(|keyboard| keyboard.active_keymap.to_string())
+        })
+        .unwrap_or_else(|| "unknown".to_string());
+
+    let num_layouts = match hyprland::keyword::Keyword::get("input:kb_layout") {
+        Ok(layouts) => {
+            let layout_str = layouts.value.to_string();
+            let layout_list: Vec<&str> = layout_str.split(',').collect();
+            layout_list.len()
+        }
+        Err(_) => 0,
+    };
+
+    Layout {
+        num_layouts,
+        active,
+    }
+}
+#[derive(Debug, Clone)]
+pub struct Layout {
+    num_layouts: usize,
+    active: String,
+}
+
+pub struct KeyboardLayout {
+    layout: Layout,
+}
+
+impl Default for KeyboardLayout {
+    fn default() -> Self {
+        Self {
+            layout: get_layout(),
+        }
+    }
+}
```

Comment:
This could be simplified into (feel free to change names)
```suggestion
#[derive(Debug, Clone)]
pub struct KeyboardLayout {
    multiple_layout: bool,
    active: String,
}

impl Default for KeyboardLayout {
    fn default() -> Self {
        Self {
            multiple_layout: get_multiple_layout_flag(),
            active: get_active_layout(),
        }
    }
}
```